src/controllers/controller.bar.js

```diff
@@ -383,11 +391,17 @@ module.exports = function(Chart) {
 			var meta = me.getMeta();
 			var scale = me.getValueScale();
 			var datasets = chart.data.datasets;
-			var value = scale.getRightValue(datasets[datasetIndex].data[index]);
+			// float-bar support, if y arguments are array function will use top - bottom value to calculate bar height
+			var value = scale.parseValue(datasets[datasetIndex].data[index]);
 			var stacked = scale.options.stacked;
 			var stack = meta.stack;
+			// float-bar support, if y arguments are array function will use proper value as bar start point
 			var start = 0;
-			var i, imeta, ivalue, base, head, size;
+			if (value.min !== value.max) {
+				start = value.val > 0 ? value.min : value.max;
+			}
+
+			var i, imeta, ivalue, base, head, size, yStackValue;
```

Comment:
move this line before the `if` statement. for minification purposes we try to keep all the variable declarations next to each other

Reply:
done